components/app.js

```diff
@@ -43,65 +40,30 @@ class App {
     this.gradeTable.identityTransfer(this.identityNum);
   }
   createGrade(name, course, grade) {
-    $.ajax('https://sgt.lfzprototypes.com/api/grades', {
-        type: 'POST',
-        error: this.handleCreateGradeError,
-        success: this.handleCreateGradeSuccess,
-        headers: {
-                  'X-Access-Token': 'qCk8Xlz9'
-                },
-        data: {
-                name: name,
-                course: course,
-                grade: grade
-              }
-    });
-  }
-  handleCreateGradeError(error) {
-    console.error("Grade creation failed: Network connection could not be established");
-  }
-  handleCreateGradeSuccess() {
-    this.getGrades();
-  }
-  deleteGrade(id) {
-    $.ajax('https://sgt.lfzprototypes.com/api/grades/' + id, {
-        type: 'DELETE',
-        error: this.handleDeleteGradeError,
-        success: this.handleDeleteGradeSuccess,
-        headers: {
-                  'X-Access-Token': 'qCk8Xlz9'
-                },
-
+      this.array.push( {
+                         id: this.array.length - 1,
+                         name: name,
+                         course: course,
+                         grade: Number(grade)
       });
+    this.refresh();
+    }
+  deleteGrade(rowNum) {
+    this.array.splice(rowNum, 1);
+    this.refresh();
   }
-  handleDeleteGradeError(error) {
-    console.error("Delete grade failed.");
-  }
-  handleDeleteGradeSuccess() {
-    this.getGrades();
-  }
+
   changeGrade(name, course, grade) {
-    $.ajax('https://sgt.lfzprototypes.com/api/grades/' + this.id, {
-        type: 'PATCH',
-        error: this.handleChangeGradeError,
-        success: this.handleChangeGradeSuccess,
-        headers: {
-          'X-Access-Token': 'qCk8Xlz9'
-        },
-        data: {
-                name: name,
-                course: course,
-                grade: grade
-        }
-      });
-  }
-  handleChangeGradeError(error) {
-    console.error("Change grade failed.");
-  }
-  handleChangeGradeSuccess() {
-    this.getGrades();
+    this.array[this.id].name = name;
+    this.array[this.id].course = course;
+    this.array[this.id].grade = Number(grade);
+    this.refresh();
   }
```

Comment:
While this does greatly reduce network communication, the API still needs to be sent grade create / update / delete requests, otherwise the app loses any changes on a refresh. What should happen is somewhere in the middle of the old and the current functionality:
- The front-end _does_ send `POST` / `PATCH` / `DELETE` requests to the API, so the primary source of grades is updated.
- However, the front-end _does not_ perform another `GET` request afterward to get the entire list of grades again: it just modifies it's local copy of the grades based on the response of the `POST` / `PATCH` / `DELETE` request.